mls-rs/src/group/mod.rs

```diff
@@ -516,10 +520,17 @@ where
         // index represent the index of this node among the leaves in the tree, namely the index of
         // the node in the tree array divided by two.
         let self_index = public_tree
-            .find_leaf_node(&key_package_generation.key_package.leaf_node)
+            .find_leaf_node(&key_package.leaf_node)
             .ok_or(MlsError::WelcomeKeyPackageNotFound)?;
 
-        let used_key_package_ref = key_package_generation.reference;
+        #[cfg(not(feature = "last_resort_key_package_ext"))]
+        let is_last_resort = false;
+        #[cfg(feature = "last_resort_key_package_ext")]
+        let is_last_resort = key_package
+            .extensions
+            .has_extension(LastResortKeyPackageExt::extension_type());
+        // Delete the key just used if this is not a last-resort key package.
+        let used_key_package_ref = (!is_last_resort).then_some(key_package_generation.reference);
```

Comment:
I may be opening a can of worms here, but would it be useful to add a function on SQLite storage to delete all last resort packages? This way we could rotate them before they expire?

Reply:
Yeah, that could make sense.

I guess the behavior here is left up to the implementation by the RFC. I'm happy to extend the SQLite layer with such a method, perhaps in a later PR if people feel it's needed?

Reply:
That would be great, but remember that we need backwards compatibility on the DB :/

Reply:
Given that we have [`SqLiteKeyPackageStorage::delete`](https://docs.rs/mls-rs/latest/mls_rs/storage_provider/sqlite/storage/struct.SqLiteKeyPackageStorage.html#method.delete) already, perhaps it would be more appropriate to

- add a way to iterate over the key packages
- add a method to determine if a key package is a last resort package

Reply:
Let me resolve this for now — this PR is about the core functionality. Deleting non-expired key packages seems like a potential next step to me.